command/operator_api.go

```diff
@@ -142,7 +144,15 @@ func (c *OperatorAPICommand) Run(args []string) int {
 	stat, _ := os.Stdin.Stat()
 	if (stat.Mode() & os.ModeCharDevice) == 0 {
 		verbose("* Reading request body from stdin.")
-		c.body = os.Stdin
+
+		// Load stdin into a *bytes.Reader so that http.NewRequest can set the
+		// correct Content-Length value.
+		b, err := ioutil.ReadAll(os.Stdin)
+		if err != nil {
+			c.Ui.Error(fmt.Sprintf("Error reading stdin: %v", err))
+			return 1
+		}
+		c.body = bytes.NewReader(b)
```

Comment:
I'm not sure if this is actually the best way to handle this, so feel free to suggest any alternative implementation 😅

Reply:
Hm, a generic HTTP client intended to work with any HTTP server should use chunked TransferEncoding here...

...but that's not the world this code lives in! This is a client for the Nomad API, and the Nomad API (1) peeks at ContentLength in some places as you note, and (2) has no APIs that expect large request bodies.

So this seems like a good approach and has the added benefit of failing early if stdin breaks.